WalletWasabi/Stores/IndexStore.cs

```diff
@@ -69,17 +82,83 @@ public async Task InitializeAsync(CancellationToken cancellationToken)
 		{
 			cancellationToken.ThrowIfCancellationRequested();
 
+			// Migration code.
+			if (RunMigration)
+			{
+				await MigrateToSqliteNoLockAsync(cancellationToken).ConfigureAwait(false);
+			}
+
 			await InitializeFiltersNoLockAsync(cancellationToken).ConfigureAwait(false);
 		}
 	}
 
+	private async Task MigrateToSqliteNoLockAsync(CancellationToken cancellationToken)
+	{
+		int i = 0;
+
+		// Helps to avoid UI freezing.
+		await Task.Delay(5_000, cancellationToken).ConfigureAwait(false);
```

Comment:
This help only because it is giving back the execution to the caller thread. When it returns here it might be running on a separate task, might not - it is not guaranteed. If the UI thread will get the task it will still freeze the UI. 

Because you are not having an async method for `MigrateToSqliteNoLockAsync` (perf reasons cACK) I suggest removing the Delay and changing it back to the sync method.

Then call it with 

`await Task.Run(MigrateToSqliteNoLockAsync);`